pkg/controller/eventing/subscription/reconcile.go

```diff
@@ -247,11 +262,72 @@ func (r *reconciler) fetchObjectReference(namespace string, ref *corev1.ObjectRe
 	return resourceClient.Get(ref.Name, metav1.GetOptions{})
 }
 
-func (r *reconciler) reconcileFromChannel(namespace string, subscribable corev1.ObjectReference, callDomain string, resultDomain string, deleted bool) error {
-	glog.Infof("Reconciling From Channel: %+v call: %q result %q deleted: %v", subscribable, callDomain, resultDomain, deleted)
+func (r *reconciler) syncPhysicalFromChannel(sub *v1alpha1.Subscription) error {
+	glog.Infof("Reconciling Physical From Channel: %+v", sub)
 
+	subs, err := r.listAllChannelsWithPhysicalFrom(&sub.Status.PhysicalSubscription.From)
+	if err != nil {
+		glog.Infof("Unable to list all channels with physical from: %+v", err)
+		return err
+	}
+
+	channelable := r.createChannelable(subs)
+
+	return r.patchPhysicalFrom(sub.Namespace, sub.Status.PhysicalSubscription.From, channelable)
+}
+
+func (r *reconciler) listAllChannelsWithPhysicalFrom(physicalFrom *corev1.ObjectReference) ([]v1alpha1.Subscription, error) {
+	subs := make([]v1alpha1.Subscription, 0)
+	opts := &client.ListOptions{
+		// TODO this is here because the fake client needs it. Remove this when it's no longer
+		// needed.
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: v1alpha1.SchemeGroupVersion.String(),
+				Kind:       "Subscription",
+			},
+		},
+	}
+	ctx := context.TODO()
+	for {
```

Comment:
This doesn't actually work with controller-runtime because the cache only supports selecting on a single field (for now). See https://github.com/kubernetes-sigs/controller-runtime/blob/ddf0390deebe496cb449b1b8540031da9236afa2/pkg/cache/internal/cache_reader.go#L160.

So definitely go with the loop.